s3/cmd/main.go

```diff
@@ -16,20 +16,63 @@ package main
 
 import (
 	"fmt"
+	"os"
 
-	micro "github.com/micro/go-micro"
+	"github.com/micro/go-micro"
 	handler "github.com/opensds/multi-cloud/s3/pkg"
+	_ "github.com/opensds/multi-cloud/s3/pkg/datastore"
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/driver"
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/yig/config"
+	"github.com/opensds/multi-cloud/s3/pkg/helper"
+	"github.com/opensds/multi-cloud/s3/pkg/log"
+	"github.com/opensds/multi-cloud/s3/pkg/meta/redis"
 	pb "github.com/opensds/multi-cloud/s3/proto"
 )
 
+var logger *log.Logger
+
 func main() {
 	service := micro.NewService(
 		micro.Name("s3"),
 	)
 
-	micro.NewService()
+	service.Init(micro.AfterStop(func() error {
+		driver.FreeCloser()
+		return nil
+	}))
+
+	helper.SetupConfig()
+
+	//yig log
+	f, err := os.OpenFile(helper.CONFIG.LogPath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		panic("Failed to open log file " + helper.CONFIG.LogPath)
+	}
+	defer f.Close()
+
+	logger = log.New(f, "[yig]", log.LstdFlags, helper.CONFIG.LogLevel)
+	helper.Logger = logger
```

Comment:
please move logger initialization inside helper...to all the remaining code outside of helper, the logger is abstracted by helper. in helper, we can have methods to write the log...better still, make a separate LogHelper that ONLY manages the various logs

Reply:
Hi rhsakarpos, we have replaced the old logger with the new logger provided by multicloud, please check it again, thanks.